yt_dlp/extractor/bilibili.py

```diff
@@ -755,15 +755,24 @@ def json2srt(self, json):
             for i, line in enumerate(json['body']) if line.get('content'))
         return data
 
-    def _get_subtitles(self, ep_id):
-        sub_json = self._call_api(f'/web/v2/subtitle?episode_id={ep_id}&platform=web', ep_id)
+    def _get_subtitles(self, *, ep_id=None, aid=None):
+        params = {
+            'platform': 'web'
+        }
+        if ep_id:
+            params['episode_id'] = ep_id
+        if aid:
+            params['aid'] = aid
```

Comment:
this code is repeated in get_formats. It can be separated into a function (or better, incoperated into call_api?)

Reply:
It's not quite identical - in one call, the param is `episode_id` while in the other, it's `ep_id`. I don't like the repetition either, but I also don't want to have a flag argument controlling "do we use the long or the short param name?".

I think the way I'd like to be able to write this code is with a helper to exclude `None` from dicts, so that the similar-but-not-identical dicts can just be created inline. Still somewhat repetitive, but the repeated section is smaller/cleaner:

```
    self._call_api(
        ...,
        query=exclude_none({
            'platform': 'web',
            'episode_id': ep_id,
            'aid': aid,
        }))
...

def exclude_none(d: dict):
    return {k: v for k, v in d.items() if v is not None}
```

What are your thoughts on a helper like that? I think I've seen this kind of conditional-add-to-dict logic enough that it would make sense for something in utils.py.

Reply:
sure, u can add this to utils:
```py
def filter_dict(dct, cndn=lambda _, v: v is not None):
    return {k: v for k, v in dct.items() if cndn(k, v)}
```

Reply:
Ive pushed the function 90137ca4bea0a22afec5bc6a0a2c8ff60ea76975